File: src/atcoder/abc314/ruby/abc314_b.rb
Player = Struct.new(:id, :num, :bet)
n = gets.to_i
players = Array.new(n) do |i|
  c = gets.to_i
  a = gets.split.map(&:to_i)
  Player.new(i + 1, c, a)
end
x = gets.to_i

winner = players.select do |player|
  player.bet.include?(x)
end

if winner.empty?
  puts 0
  exit
end

mini = winner.map(&:num).min
ans = winner.select do |player|
  player.num == mini
end

puts ans.size
puts ans.map(&:id).join(" ")

File: src/atcoder/abc314/python3/abc314_b.py
from dataclasses import dataclass

@dataclass
class Player:
  id: int
  num: int
  bet: [int]

n = int(input())
players = [
  Player(i + 1, int(input()), [int(s) for s in input().split()])
  for i in range(n)
]
x = int(input())

winner = [player for player in players if x in player.bet]
if len(winner) == 0:
  print(0)
else:
  mini = min([player.num for player in winner])
  ans = [player.id for player in winner if player.num == mini]
  print(len(ans))
  print(" ".join(map(str, ans)))